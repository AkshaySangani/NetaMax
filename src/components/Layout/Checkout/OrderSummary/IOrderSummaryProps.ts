export interface IOrderSummaryProps {
  /**
   * The total discount amount.
   * @type {number}
   **/
  totalDiscountPrice: number;

  /**
   * The net total amount.
   * @type {number}
   */
  totalNetPrice: number;
}
