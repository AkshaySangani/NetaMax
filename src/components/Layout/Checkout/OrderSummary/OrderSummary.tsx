import { ReactElement } from "react";

import { CURRENCY_NAME } from "constants/localeFormatConstants";
import { formatMoney } from "utils/currencyUtils";

import { Box, Text } from "@chakra-ui/react";

import { IOrderSummaryProps } from "./IOrderSummaryProps";

/**
 * The order summary component.
 * @param {IOrderSummaryProps} props the props
 * @returns {ReactElement} the component
 */
export const OrderSummary = (props: IOrderSummaryProps): ReactElement => {
  const { totalDiscountPrice, totalNetPrice } = props;

  return (
    <Box height="120px">
      <Box display="flex" bg="#3CE88B" alignItems="center" justifyContent="center" height="51px">
        <Box fontSize={{ base: "md", md: "md" }} display="flex">
          Est√°s ahorrando
          <Text padding="0 0.25em" fontWeight="bold">
            {formatMoney(totalDiscountPrice, CURRENCY_NAME)}
          </Text>
          en esta orden
        </Box>
      </Box>
      <Box display="flex" justifyContent="space-between" m="0 16px" marginTop="20px">
        <Text> Sub-Total:</Text>
        {formatMoney(totalNetPrice > 0 ? totalNetPrice : 0, CURRENCY_NAME)}
      </Box>
    </Box>
  );
};
