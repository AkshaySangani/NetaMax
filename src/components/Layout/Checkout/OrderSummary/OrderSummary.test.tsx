import { render, cleanup } from "@testing-library/react";
import { OrderSummary } from "./OrderSummary";
import { IOrderSummaryProps } from "./IOrderSummaryProps";

describe("OrderSummary Component", () => {
  let mock: IOrderSummaryProps;

  beforeEach(() => {
    mock = {
      totalDiscountPrice: 50,
      totalNetPrice: 1000,
    };
  });

  afterEach(() => {
    cleanup();
  });

  test("OrderSummary should be render correctly", () => {
    const { container } = render(<OrderSummary {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("Total price should be 0 if the totalNetPrice is 0", () => {
    mock = {
      totalDiscountPrice: 50,
      totalNetPrice: 0,
    };
    const { container } = render(<OrderSummary {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
