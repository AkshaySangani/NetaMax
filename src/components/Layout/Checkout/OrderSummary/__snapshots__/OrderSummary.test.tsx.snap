// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummary Component OrderSummary should be render correctly 1`] = `
<div>
  <div
    class="css-6gxhc5"
  >
    <div
      class="css-1676o8c"
    >
      <div
        class="css-sf19nn"
      >
        Estás ahorrando
        <p
          class="chakra-text css-105v11o"
        >
          $50.00
        </p>
        en esta orden
      </div>
    </div>
    <div
      class="css-17m8p39"
    >
      <p
        class="chakra-text css-0"
      >
         Sub-Total:
      </p>
      $1,000.00
    </div>
  </div>
</div>
`;

exports[`OrderSummary Component Total price should be 0 if the totalNetPrice is 0 1`] = `
<div>
  <div
    class="css-6gxhc5"
  >
    <div
      class="css-1676o8c"
    >
      <div
        class="css-sf19nn"
      >
        Estás ahorrando
        <p
          class="chakra-text css-105v11o"
        >
          $50.00
        </p>
        en esta orden
      </div>
    </div>
    <div
      class="css-17m8p39"
    >
      <p
        class="chakra-text css-0"
      >
         Sub-Total:
      </p>
      $0.00
    </div>
  </div>
</div>
`;
